src/rust/engine/dep_inference/build.rs

```diff
@@ -26,10 +26,18 @@
 #![allow(clippy::mutex_atomic)]
 
 use sha2::{Digest, Sha256};
+use std::collections::HashMap;
 use std::{collections::HashSet, io::Write, path::Path};
 use std::{env, fs};
 use walkdir::WalkDir;
 
+/// The tree-sitter interfaces don't have nice constants that allow us to reference their magic numbers by name.
+/// We generate those constants here.
+/// Tree-sitter grammars don't have to give symbols unique names (in `ts_symbol_names`),
+/// and there can be multiple symbols mapped to the same name.
+/// For example, they might map both `block` and `_match_block` to "block" because one of those in internal
+/// For most names, there will only be 1 symbol; for those, we create a const u16 for convenience.
+/// For the names with multiple symbols, we generate a const array (hashmaps would have been nice, but I couldn't figure out how to make them const)
```

Comment:
I think a `lazy_static!` is the way to go for hashmaps if you still want to do that

Reply:
hmm, I can't seem to get it working, getting problems with using the object model of rust:
```
error: struct is not supported in `trait`s or `impl`s
  --> pants/src/rust/engine/target/debug/build/dep_inference-ceee0fd7ceefb634/out/python/constants.rs:42:3
   |
42 |   lazy_static! { pub static ref BLOCK: HashSet<u16> = HashSet::from([157, 133]); }
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider moving the struct out to a nearby module scope
   = note: this error originates in the macro `lazy_static` (in Nightly builds, run with -Z macro-backtrace for more info)
```
I'm not a real rustacean and I don't understand what's going on. I think I'll drop it and let some cooler crabs handle it.

Reply:
Honestly, ChatGPT does have some helpful hints when asked about the crustacean formations

Reply:
yeah, It's got me this far. It hasn't given a useful suggestion that I feel is a major improvement, either "use a function" or "pull it out of the impl". If I knew more Rust I might be able to prompt it better, but alas.